Src/IronPython.Modules/_codecs.cs

```diff
@@ -248,51 +250,58 @@ public static PythonTuple mbcs_encode(CodeContext/*!*/ context, [NotNull]string
         #endregion
 
         #region Raw Unicode Escape Encoding Functions
+#if FEATURE_ENCODING
```

Comment:
Unrelated to this PR, but I think we can drop the whole `FEATURE_ENCODING` from the codebase. It was meant for platforms such as Silverlight but I think we can safely stop supporting those.

Reply:
There are more compilation [features defined](https://github.com/IronLanguages/ironpython3/blob/master/Documentation/feature-symbols.md) than just `FEATURE_ENCODING`. Maybe it is a long shot but maybe one day IronPython will be used in a (relatively) resource constrained environment, like a mobile platform, Raspberry Pi, or IoT device, and the ability to exclude all that encoding stuff will be handy. It is already in place.

The whole list of feature symbols will need to be reviewed, updated and documented at some point. For encodings, I just need to get in a habit to build with this feature disabled, to catch compiler errors. How about defining a property, say `Bare` that would produce a build with all features disabled? No testing, just compilation.

Reply:
Indeed, there are many feature symbols defined, most of them are there due to missing APIs in one platform or another. I had done a first pass at eliminating some feature symbols https://github.com/IronLanguages/ironpython2/issues/200 although I had need very tolerant. For ipy3 I would lean towards eliminating any feature switch for features part of .NET Standard 2.0 (e.g. FEATURE_DBNULL, FEATURE_APPLICATION_EXCEPTION, etc).

I guess `FEATURE_ENCODING` could be open to discussion. According to the above criterion it could be eliminated, however limiting to "base" encodings could be an option. We should however review what gets excluded by this symbol, for example it seems like ignoring `errors` is not a good idea? Anyway, probably something for another PR.

"Constrained environments" is indeed a bit of a long shot, I just ran the test suite on a Raspberry pi. 😄

Reply:
After thinking about it for a few days, I now agree that `FEATURE_ENCODING` as it is can better be dropped. Something for a separate PR.